x-pack/filebeat/input/http_endpoint/handler.go

```diff
@@ -113,3 +124,14 @@ func httpReadJSON(body io.Reader) (objs []common.MapStr, status int, err error)
 	}
 	return objs, 0, nil
 }
+
+func getIncludedHeaders(r *http.Request, headerconf []string) (includedHeaders common.MapStr) {
+	includedHeaders = common.MapStr{}
+	for _, header := range headerconf {
+		h := r.Header.Get(header)
```

Comment:
Maybe accessing the map directly would be better for the purpose here, since `Get` only returns the first value for the header, but http headers can contain multiple (https://golang.org/pkg/net/http/#Header.Get).

```suggestion
		h := r.Header[textproto. CanonicalMIMEHeaderKey(header)] // CanonicalMIMEHeaderKey will set the right casing to get the header, Get does this automatically
```

Reply:
Applied the changes, will have to take into account that the value of the header will now be arrays, but that should still work.